tao_compiler/mlir/disc/transforms/fusion_utils.cc

```diff
@@ -1082,14 +1100,24 @@ bool BaseFusionStrategy::tryFuse(ShapeAnalysis& shapeAnalysis,
     return true;
   }
 
-  Value ref_shape = getEffectiveShape(target, results[0]);
-  if (!llvm::all_of(results, [&](Value result) {
-        Value shape = getEffectiveShape(target, result);
-        return checkSameShape(lhs, rhs, target)
-                   ? shapeAnalysis.isShapeEqual(ref_shape, shape)
-                   : shapeAnalysis.isSameNumElements(ref_shape, shape);
-      })) {
-    return false;
+  if (rhs.getFusionType() != FusionType::kWhere &&
+      lhs.getFusionType() != FusionType::kWhere) {
+    Value ref_shape = getEffectiveShape(target, results[0]);
+    if (!llvm::all_of(results, [&](Value result) {
+          Value shape = getEffectiveShape(target, result);
+          return checkSameShape(lhs, rhs, target)
+                     ? shapeAnalysis.isShapeEqual(ref_shape, shape)
+                     : shapeAnalysis.isSameNumElements(ref_shape, shape);
+        })) {
+      return false;
+    }
+  } else {
+    // TODO(lanbo.llb): make a more general rule for kWhere fusion
+    if (results.size() > 2) {
```

Comment:
Only check the number of results less than 2 can not make sure WhereOp is the root op？